camayoc/tests/qcs/cli/test_sources.py

```diff
@@ -777,6 +781,117 @@ def test_clear(isolated_filesystem, qpc_server_config, source_type):
     qpc_source_show.close()
 
 
+def test_clear_with_scans(isolated_filesystem, qpc_server_config, source_type):
+    """Clear a source which is used in scans.
+
+    :id: b10435c0-db94-4431-a580-575dd7db4ced
+    :description: Clear a source entry by entering the ``--name`` of an
+        already created entry.
+    :steps: Run ``qpc source clear --name <name>``
+    :expectedresults: The source entry is removed.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    hosts = '127.0.0.1'
+    port = default_port_for_source(source_type)
+    cred_add(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
```

Comment:
There is a source_add utility function that you may want to use. Not a blocker to get this PR merged

Reply:
That's a great suggestion.  Maybe a separate PR.  The other existing tests use the style above.      I can make a pass to update all source tests to use the util.